conans/client/generators/cmake_find_package.py

```diff
@@ -43,52 +48,297 @@ class CMakeFindPackageGenerator(Generator):
-                                                      deps_names=deps_names)
-
-        # The find_transitive_dependencies block
-        find_dependencies_block = ""
-        if dep_cpp_info.public_deps:
-            # Here we are generating FindXXX, so find_modules=True
-            f = find_transitive_dependencies(public_deps_names, find_modules=True)
-            # proper indentation
-            find_dependencies_block = ''.join("        " + line if line.strip() else line
-                                              for line in f.splitlines(True))
-
-        tmp = self.find_template.format(name=name, version=dep_cpp_info.version,
-                                        find_libraries_block=find_libraries_block,
-                                        find_dependencies_block=find_dependencies_block,
-                                        macros_and_functions=macros_and_functions)
-        return tmp
+    def _get_components(self, pkg_name, pkg_findname, cpp_info, reverse=True):
+        find_package_components = []
+        for comp_name, comp in self.sorted_components(cpp_info).items():
+            comp_findname = self._get_name(cpp_info.components[comp_name])
+            deps_cpp_cmake = DepsCppCmake(comp)
+            deps_cpp_cmake.public_deps = self._get_component_requires(pkg_name, pkg_findname, comp)
+            find_package_components.append((comp_findname, deps_cpp_cmake))
+        if reverse:
+            find_package_components.reverse()  # From the less dependent to most one
+        return find_package_components
+
+    def _get_component_requires(self, pkg_name, pkg_findname, comp):
+        comp_requires_findnames = []
+        for require in comp.requires:
+            if COMPONENT_SCOPE in require:
+                comp_require_pkg_name = require[:require.find(COMPONENT_SCOPE)]
+                if comp_require_pkg_name not in self.deps_build_info.deps:
+                    raise ConanException("Component '%s' not found: '%s' is not a package "
+                                         "requirement" % (require, comp_require_pkg_name))
+                comp_require_pkg = self.deps_build_info[comp_require_pkg_name]
+                comp_require_pkg_findname = self._get_name(comp_require_pkg)
+                comp_require_comp_name = require[require.find(COMPONENT_SCOPE)+len(COMPONENT_SCOPE):]
+                if comp_require_comp_name in self.deps_build_info.deps:
+                    comp_require_comp_findname = comp_require_pkg_findname
+                elif comp_require_comp_name in self.deps_build_info[comp_require_pkg_name].components:
+                    comp_require_comp = comp_require_pkg.components[comp_require_comp_name]
+                    comp_require_comp_findname = self._get_name(comp_require_comp)
+                else:
+                    raise ConanException("Component '%s' not found in '%s' package requirement"
+                                         % (require, comp_require_pkg_name))
```

Comment:
IMO, this is not needed, everything is validated in the `cpp_info` object

Reply:
This one is not validated in the model